### DISCUSSION TITLE
Keeping integrations more in sync with latest mermaid version?

### DISCUSSION THREAD

**Discussion by andorsk** [mermaid-js/mermaid#3999] [General]
I love the mermiad project and I use it all the time. One of my favorite parts about the library is that HackMD and Github both natively render mermaid, so it's unfortunate when Github fails to support a feature that is available in a recent version of mermaid. 

I have two questions to bring into the discussion: 

1. How do we keep Github and HackMD integrations more aligned with the latest and greatest version of mermaid long term?
2. Is there any way we could (short term), upgrade the integrations to the Github ( and HackMD ) markdown processor? 

See the below as an example of compatibility issues between mermaid.live and Github's render. 

## Example

**The following markdown:** 
```
classDiagram

    class ClassA{
        +A
    }

    note for ClassA "This is a test"
  ```

**Rendered over Github**

```mermaid
classDiagram

    class ClassA{
        +A
    }

    note for ClassA "This is a test"
```

**Rendered over Github after removing note (note was added in late 2022)**

```mermaid
classDiagram

    class ClassA{
        +A
    }  
```

**Or when rendered over mermaid.live it works with the note:**

<img width="1438" alt="image" src="https://user-images.githubusercontent.com/8604639/212628315-1ee8c77f-1fbd-45f0-bc79-5647b21d3d8b.png">

As you can see, notes are not supported in github right now, but are supported in mermaid. This is an annoying problem, b/c there is not feature parity between designing in mermaid.live and a github render.

**Comment by Yokozuna59** [comment]
What would you suggest? I don't think we could resolve this; only tools created by us could we do something about, i.e., `mermaid.live`. Some of those tools are closed source (we don't access in the repo), how could be even create a PR to bump the version?

We could suggest adding some workflows, e.i., dependabot, to notify those contributors to update the version. But it wouldn't matter if they didn't want to update it.

---

I'll close this discussion for now, feel free to discuss more or reopen it my answer didn't help.
